runtime/cudaq/platform/default/rest_server/helpers/RestRemoteServer.cpp

```diff
@@ -220,10 +355,12 @@ class RemoteRestRuntimeServer : public cudaq::RemoteRuntimeServer {
     return uniqueJit;
   }
 
-  void invokeMlirKernel(std::unique_ptr<MLIRContext> &contextPtr,
-                        std::string_view irString,
-                        const std::vector<std::string> &passes,
-                        const std::string &entryPointFn) {
+  void
+  invokeMlirKernel(std::unique_ptr<MLIRContext> &contextPtr,
+                   std::string_view irString,
+                   const std::vector<std::string> &passes,
+                   const std::string &entryPointFn, std::size_t numTimes = 1,
+                   std::function<void(std::size_t)> postExecCallback = {}) {
```

Comment:
I didn't notice that fact until you pointed it out. I googled it and it's governed by the "comple-class context" concept.

> A complete-class context of a class is a
> (6.1) **function body** ([dcl.fct.def.general]),
> (6.2) default argument,
> (6.3) noexcept-specifier ([except.spec]), or
> (6.4) default member initializer
> within the member-specification of the class.
> .....
> A class is considered a completely-defined object type ([basic.types]) (or complete type) at the closing } of the class-specifier. **The class is regarded as complete within its complete-class contexts**; otherwise it is regarded as incomplete within its own class member-specification.

So I think in this case, the declaration of `invokeMlirKernel` is required to be processed before the member function bodies where it's used.

Reply:
That's definitely interesting. I guess it would throw a linker error if it accidentally assumed the wrong function prototype (because the C++ name mangling would cause the incorrectly assumed wrong function prototype to not resolve.) So I'm good with it as is.